### PR TITLE
Fix local TensorFlow example build

### LINKED ISSUE #476: Error building tensorflow example locally
Not sure why, but this started failing for me locally when running the following command in the latest Docker image:

```
./scripts/docker_run ./scripts/build_examples
```

logs:

```log
+ bazel build --config=emscripten //examples/tensorflow/module/cpp:all
INFO: Writing tracer profile to '/.cache/bazel/_bazel_tzn/b6bc0828d5b43c7085b650e0698bf388/command.profile.gz'
INFO: Analyzed target //examples/tensorflow/module/cpp:tensorflow.wasm (0 packages loaded, 0 targets configured).
INFO: Found 1 target...
ERROR: /.cache/bazel/_bazel_tzn/b6bc0828d5b43c7085b650e0698bf388/external/org_tensorflow/tensorflow/lite/c/BUILD:6:1: C++ compilation of rule '@org_tensorflow//tensorflow/lite/c:c_api_internal' failed (Exit 1)
Traceback (most recent call last):
  File "/usr/local/emsdk/upstream/emscripten/emcc.py", line 41, in <module>
    import emscripten
  File "/usr/local/emsdk/upstream/emscripten/emscripten.py", line 24, in <module>
    from tools import shared
  File "/usr/local/emsdk/upstream/emscripten/tools/shared.py", line 300, in <module>
    generate_config(EM_CONFIG, first_time=True)
  File "/usr/local/emsdk/upstream/emscripten/tools/shared.py", line 226, in generate_config
    open(abspath, 'w').write(config_file)
IOError: [Errno 2] No such file or directory: '/.cache/bazel/_bazel_tzn/b6bc0828d5b43c7085b650e0698bf388/execroot/oak/~/.emscripten'
Target //examples/tensorflow/module/cpp:tensorflow.wasm failed to build
Use --verbose_failures to see the command lines of failed build steps.
INFO: Elapsed time: 0.933s, Critical Path: 0.28s
INFO: 0 processes.
FAILED: Build did NOT complete successfully
```

### PR DESCRIPTION
**Pull request by ipetr0v**
This change fixes TensorFlow example build in non-root Docker containers.

Fixes #476

### CODE REVIEW

File: scripts/docker_run [hunk 1]
```diff
@@ -32,6 +32,7 @@ docker_run_flags=(
   '--rm'
   "--user=$DOCKER_UID:$DOCKER_GID"
   "--env=USER=$DOCKER_USER"
+  "--env=HOME=/tmp"
```
[the remark this answers is not in the document]
Comment (ipetr0v) [reply scripts/docker_run]: Deleted in favor of settings `EM_CONFIG` in the `emcc.sh`

File: Dockerfile [hunk 2]
```diff
@@ -90,11 +90,15 @@ RUN rm ${GOLANG_TEMP}
...
-ARG EMSCRIPTEN_COMMIT=369013943283939412fb2807bb0d2ded8ebd5a9e
-ARG EMSCRIPTEN_SHA256=0d192691e3186cf14833ab861a19262996a6bf69b62cb205d8662068d57f68ab
+ARG EMSCRIPTEN_VERSION=1.39.6
+ARG EMSCRIPTEN_COMMIT=6bfbe2a7da68e650054af2d272d2b79307a6ad72
+ARG EMSCRIPTEN_SHA256=aa4c3b8f23fd26363f98207674bffcc138105c621c6c8bf12175f6aab1231357
 ARG EMSCRIPTEN_DIR=/usr/local/emsdk
 ARG EMSCRIPTEN_TEMP=/tmp/emscripten.zip
+ENV PATH "${EMSCRIPTEN_DIR}:${EMSCRIPTEN_DIR}/upstream/emscripten:${PATH}"
+ENV EMSDK "${EMSCRIPTEN_DIR}"
+ENV EM_CONFIG '/tmp/.emscripten'
```
[the remark this answers is not in the document]
Comment (ipetr0v) [reply Dockerfile]: I have deleted setting `$HOME` to `/tmp`, but we still need to have a directory, where docker user will be able to create Emscripten cache files.
But I can change it from `/tmp` to something else.

File: Dockerfile [hunk 3]
```diff
@@ -103,6 +107,9 @@ RUN rm ${EMSCRIPTEN_TEMP}
 RUN cd ${EMSCRIPTEN_DIR} \
     && ./emsdk install ${EMSCRIPTEN_VERSION} \
     && ./emsdk activate ${EMSCRIPTEN_VERSION}
-ENV PATH "${EMSCRIPTEN_DIR}:${EMSCRIPTEN_DIR}/upstream/emscripten:${PATH}"
-ENV EMSDK "${EMSCRIPTEN_DIR}"
-ENV EM_CONFIG "/root/.emscripten"
+# Emscripten config file is automatically generated in `/root/`, so we need to allow access to it
```
Comment (tiziano88) [inline Dockerfile]: Is it generated in `/root` or in `$HOME`? I know they are the same here, but I think `$HOME` is more accurate, right?

File: Dockerfile [hunk 4]
```diff
@@ -103,6 +107,9 @@ RUN rm ${EMSCRIPTEN_TEMP}
...
+# and put it in a directory with write access, so `emcc` will be able to create `.emscripten_cache.lock`
+RUN cp /root/.emscripten "${EM_CONFIG}" \
+    && chmod ugo+rw "${EM_CONFIG}" \
+    && mkdir -p "${EM_CACHE}" \
```
Comment (tiziano88) [inline Dockerfile]:
```suggestion
    && mkdir --parents "${EM_CACHE}" \
```

**Pull request merged**